src/_internal/utils.ts

```diff
@@ -40,3 +41,20 @@ export const asyncEmpty = async function* () {};
 
 export const isNotNullable = <T>(a: T): a is NonNullable<T> =>
   a !== null && a !== undefined;
+
+export const isPromise = <T>(a: Promise<T> | T): a is Promise<T> => {
+  if (a instanceof Promise) {
+    return true;
+  }
+
+  if (
+    a !== null &&
+    typeof a === "object" &&
+    typeof (a as any).then === "function" &&
+    typeof (a as any).catch === "function"
```

Comment:
It seems to be necessary to discuss at what level the judgment of the thenable object should be decided.

In the case of co, only the existence of the then function is judged.
https://github.com/tj/co/blob/master/index.js#L198

Reply:
I also think this level would be appropriate.